packages/react/src/components/Space.tsx

```diff
@@ -1,160 +1,26 @@
-import { Flatfile } from '@flatfile/api'
-import { ISpace, SpaceComponent } from '@flatfile/embedded-utils'
-import React, { JSX, useEffect, useState } from 'react'
-import { useCreateListener } from '../hooks/useCreateListener'
-import { addSpaceInfo } from '../utils/addSpaceInfo'
-import { authenticate } from '../utils/authenticate'
-import ConfirmModal from './ConfirmCloseModal'
-import {
-  getContainerStyles,
-  getIframeStyles,
-} from './embeddedStyles'
-import './style.scss'
+import FlatfileContext from './FlatfileContext'
+import React, { useCallback, useContext } from 'react'
+import type { Flatfile } from '@flatfile/api'
+import { useDeepCompareEffect } from '../utils/useDeepCompareEffect'
 
 /**
  * @name Space
- * @description Flatfile Embedded Space component
+ * @description Flatfile Embedded SpaceWrapper component 
  * @param props
  */
 
-const Space = ({
-  spaceId,
-  spaceUrl,
-  accessToken,
-  handleCloseInstance,
-  ...props
-}: SpaceComponent & ISpace & {handleCloseInstance: () => void}): JSX.Element | null => {
-  if (spaceId && spaceUrl && accessToken) {
-    return (
-      <SpaceContents
-        spaceId={spaceId}
-        spaceUrl={spaceUrl}
-        accessToken={accessToken}
-        handleCloseInstance={handleCloseInstance}
-        {...props}
-      />
-    )
-  }
-  return null
-}
-
-export const SpaceContents = (
-  props: ISpace & { spaceId: string; spaceUrl: string; accessToken: string; handleCloseInstance: () => void }
-): JSX.Element => {
-  const [showExitWarnModal, setShowExitWarnModal] = useState(false)
-  const {
-    spaceId,
-    spaceUrl,
-    listener,
-    accessToken,
-    closeSpace,
-    iframeStyles,
-    mountElement = 'flatfile_iFrameContainer',
-    exitText = 'Are you sure you want to exit? Any unsaved changes will be lost.',
-    exitTitle = 'Close Window',
-    exitPrimaryButtonText = 'Yes, exit',
-    exitSecondaryButtonText = 'No, stay',
-    apiUrl = 'https://platform.flatfile.com/api',
-    displayAsModal = true,
-    handleCloseInstance
-  } = props
-
-  const { dispatchEvent } = useCreateListener({ listener, accessToken, apiUrl })
+export const Space = (props: {
+  config: Flatfile.SpaceConfig & { id?: string }
+  children?: React.ReactNode
+}) => {
+  const { config, children } = props
+  const { updateSpace, createSpace } = useContext(FlatfileContext)
 
-  const handlePostMessage = (event: any) => {
-    const { flatfileEvent } = event.data
-    if (!flatfileEvent) return
-    if (
-      flatfileEvent.topic === 'job:outcome-acknowledged' &&
-      flatfileEvent.payload.status === 'complete' &&
-      flatfileEvent.payload.operation === closeSpace?.operation
-    ) {
-      closeSpace?.onClose({})
-    }
-    dispatchEvent(flatfileEvent)
-  }
+  const callback = useCallback(() => {
+    updateSpace(config)
```

Comment:
This clean new lens has me thinking about possibly gracefully handling space create/update failure here - maybe based on a debug value for what to display 💭